packages/mui-base/src/NumberField/NumberFieldContext.ts

```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+import type { UseNumberFieldReturnValue } from '../useNumberField/useNumberField.types';
+import { NumberFieldOwnerState } from './NumberField.types';
+
+export type NumberFieldContextValue = UseNumberFieldReturnValue & {
+  ownerState: NumberFieldOwnerState;
+};
+
+export const NumberFieldContext = React.createContext<NumberFieldContextValue | null>(null);
```

Comment:
In the grid, I tend to use `undefined` because it's semantically good, it's the default empty value for javascript values, and it's less verbose for refs (don't need the `| null` type).